### PR TITLE
Feature/paragraphs

### PR DESCRIPTION
**Pull request by cableman**
This should not be merged it's only to use github for internal review.

### CODE REVIEW

File: modules/ding_paragraphs/js/ding_paragraphs.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            match = match.replace('reservation', 'availability').replace(':', '');
+            Drupal.settings.ding_availability[match] = [id];
+          }
+        }
+
+        // Ensure that behaviors are attached to the new content.
+        Drupal.attachBehaviors($('.paragraphs-block .view-mode-paragraphs-teaser'));
+      });
+    }}
+})(jQuery);
```
Comment (rimi-itk) [inline modules/ding_paragraphs/js/ding_paragraphs.js]: Missing newline.

File: modules/ding_paragraphs/js/ding_paragraphs_form.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            }
+            else {
+              $(".field-type-paragraphs .field-multiple-drag").show();
+            }
+          }
+        })
+      });
+    }
+  }
+})(jQuery);
```
Comment (rimi-itk) [inline modules/ding_paragraphs/js/ding_paragraphs_form.js]: Missing newline.

File: modules/ding_paragraphs/js/ding_paragraphs_form.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      'use strict';
+
+      $('.field-type-paragraphs').once(function () {
+        // Limit the number of times to act on DOMSubtreeModified.
+        var active = false;
+        $(this).on('DOMSubtreeModified', function () {
+          if (!active) {
+            active = true;
+            // Hide dragging if an item is not collapsed.
+            if($("input[name$='collapse_button']").length > 0) {
```
Comment (rimi-itk) [inline modules/ding_paragraphs/js/ding_paragraphs_form.js]: Missing space after `if`.

File: themes/ddbasic/ddbasic.info [hunk 4]
```diff
@@ -98,7 +99,7 @@
 
   scripts[] = scripts/ting-object/ting-object.js
 
-  scripts[] = scripts/view.js
+;  scripts[] = scripts/view.js
```
Comment (rimi-itk) [inline themes/ddbasic/ddbasic.info]: Remove unused code.

File: themes/ddbasic/sass/components/node/event.scss [hunk 5]
```diff
@@ -206,6 +206,10 @@
         // Tablet
         @include media($tablet) {
           @include span-columns(6);
+
+          &.has-paragraphs {
+            width:100%;
```
Comment (rimi-itk) [inline themes/ddbasic/sass/components/node/event.scss]: Add space after `:` (for consistency with rest of code).

File: themes/ddbasic/sass/components/node/event.scss [hunk 6]
```diff
@@ -251,6 +255,10 @@
...
```
Comment (rimi-itk) [inline themes/ddbasic/sass/components/node/event.scss]: Add space after `:` (for consistency with rest of code).

File: themes/ddbasic/sass/components/node/news.scss [hunk 7]
```diff
@@ -226,6 +226,10 @@
       // Tablet
       @include media($tablet) {
         @include span-columns(6);
+
+        &.has-paragraphs {
+          width:100%;
```
Comment (rimi-itk) [inline themes/ddbasic/sass/components/node/news.scss]: Add space after `:` (for consistency with rest of code).

File: themes/ddbasic/sass/components/node/paragraphs.scss [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        max-width: 100%;
+      }
+    }
+  }
+}
+
+// Override existing css from different files.
+.node-ding-page .field-name-field-ding-page-body > .paragraphs-text p:last-child {
+  margin-bottom: 30px;
+}
```
Comment (rimi-itk) [inline themes/ddbasic/sass/components/node/paragraphs.scss]: Missing newline.

File: modules/ding_page/ding_page.install [hunk 9]
```diff
@@ -32,3 +46,47 @@ function ding_page_update_7003() {
   variable_del('pathauto_node_ding_page_en_pattern');
   variable_del('pathauto_node_ding_page_und_pattern');
 }
+
+/**
+ * Migrate page body field to paragraph and delete field.
+ */
+function ding_page_update_7004(&$sandbox) {
+  // Make sure we have somewhere to save the old body field.
+  if (empty(field_info_instance('node', 'field_ding_page_body', 'ding_page')) || empty(field_info_instance('paragraphs_item', 'field_ding_paragraphs_text', 'ding_paragraphs_text')) || empty(field_info_instance('node', 'field_ding_page_paragraphs', 'ding_page'))) {
```
Comment (rimi-itk) [inline modules/ding_page/ding_page.install]: Consider taking the recommendation on conditions in https://www.drupal.org/docs/develop/standards/coding-standards#linelength. You will not win Most Compact Condition In Least Lines Of Code Award™ anyways.

File: themes/ddbasic/sass/components/node/paragraphs.scss (hunk 8 above)
Comment (martinyde) [reply themes/ddbasic/sass/components/node/paragraphs.scss]: Fixed

File: modules/ding_paragraphs/js/ding_paragraphs_form.js (hunk 2 above)
Comment (martinyde) [reply modules/ding_paragraphs/js/ding_paragraphs_form.js]: Fixed

File: modules/ding_paragraphs/js/ding_paragraphs.js (hunk 1 above)
Comment (martinyde) [reply modules/ding_paragraphs/js/ding_paragraphs.js]: Fixed

**Pull request closed without merging**